Mathlib/Topology/ContinuousOn.lean

```diff
@@ -205,6 +205,38 @@ theorem nhdsWithin_union (a : α) (s t : Set α) : 𝓝[s ∪ t] a = 𝓝[s] a 
   delta nhdsWithin
   rw [← inf_sup_left, sup_principal]
 
+theorem nhdsWithin_union_univ (b : α) {I₁ I₂ : Set α} (hI : Set.univ = I₁ ∪ I₂) :
+    nhds b = nhdsWithin b I₁ ⊔ nhdsWithin b I₂ := by
+  rw [← nhdsWithin_univ b, hI, nhdsWithin_union]
+
+/-- If `L` and `R` are neighborhoods of `b` within sets whose union is `Set.univ`, then
+`L ∪ R` is a neighborhood of `b`. -/
+theorem union_mem_nhds_of_mem_nhdsWithin {b : α}
+    {I₁ I₂ : Set α} (h : Set.univ = I₁ ∪ I₂)
+    {L : Set α} (hL : L ∈ nhdsWithin b I₁)
+    {R : Set α} (hR : R ∈ nhdsWithin b I₂) : L ∪ R ∈ nhds b := by
+  rw [← nhdsWithin_univ b, h, nhdsWithin_union]
+  exact ⟨mem_of_superset hL (by aesop), mem_of_superset hR (by aesop)⟩
+
+
+/-- Writing a punctured neighborhood filter as a sup of left and right filters. -/
+lemma nhds_punctured_union [LinearOrder α] {x : α} : 𝓝[≠] x = 𝓝[<] x ⊔ 𝓝[>] x := by
```

Comment:
```suggestion
lemma punctured_nhds_eq_nhdsWithin_sup_nhdsWithin [LinearOrder α] {x : α} : 𝓝[≠] x = 𝓝[<] x ⊔ 𝓝[>] x := by
```
I'm surprised that `punctured_nhds` already has precedence in Mathlib. We should probably decide what we're calling `𝓝[≠]` and similar uniformly at some point (I also see `nhdsWithin_complement_singleton`, which is very long)

Reply:
Maybe it should be called `nhds_ne`? I don't know.

Reply:
Yes, that name (and similar ones) sound good. But let's go with precedence, so that we can uniformly change at a later date.